source/Junkership.js

```diff
@@ -3,43 +3,61 @@ var Keyb = require("keyb")
 
 import Reference from "./Reference.js"
 import Projectile from "./Projectile.js"
+import Score from "./Score.js"
 
 var junkerTex = Pixi.Texture.fromImage(require("./images/blue-starship.png"))
 
 export default class Junkership extends Pixi.Sprite {
     constructor() {
         super(junkerTex)
-        this.speed = 60
-        this.score = 0
         game.playerCount++
+        this.speed = 60
+        this.score = new Score()
     }
     update(delta) {
-        if(Keyb.isDown("<up>") && this.position.y > 0) {
-            this.position.y -= this.speed * delta
+        var relativeSpeed = this.speed * delta
+        if(Keyb.isDown("<up>")) {
+            this.position.y = this.move(this.position.y - relativeSpeed)
         }
-        if(Keyb.isDown("<down>") &&
-            (this.position.y + this.height) < Reference.GAME_HEIGHT) {
-            this.position.y += this.speed * delta
+        if(Keyb.isDown("<down>")) {
+            var maxBottom = Reference.GAME_HEIGHT - this.height
+            this.position.y = this.move(this.position.y + relativeSpeed, maxBottom)
         }
-        if(Keyb.isDown("<left>") && this.position.x > 0) {
-            this.position.x -= this.speed * delta
+        if(Keyb.isDown("<left>")) {
+            this.position.x = this.move(this.position.x - relativeSpeed)
         }
-        if(Keyb.isDown("<right>") &&
-            (this.position.x + this.width) < Reference.GAME_WIDTH) {
-            this.position.x += this.speed * delta
+        if(Keyb.isDown("<right>")) {
+            var maxRight = Reference.GAME_WIDTH - this.width
+            this.position.x = this.move(this.position.x + relativeSpeed, maxRight)
         }
         if(Keyb.isJustDown("<space>")) {
-            game.addChild(new Projectile(
-                this.x + this.width,
-                this.y + this.height/2,
-                this))
+            this.shoot()
```

Comment:
I like the use of microfunctions like `.move()` and `.shoot()`, but the parameter signature for `.move()` is confusing (#imo). It could be simpler, since that method already knows the current `x` and `speed` and `width`, all it needs to know is a direction (like a unit vector) and the delta (like a magnitude)